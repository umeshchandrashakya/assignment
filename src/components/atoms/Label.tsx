import React from "react";
import styled from "styled-components";

const StyledLabel = styled.label`
font-weight: bold; 
caret-color: getColor('blue'); 
display: block;
text-align: left;
margin-right: 23%;
margin-top : 5px;
    margin-bottom : 15px;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 14px;
`;

interface Props {
  value: string;
}

const Label = (props: Props) => (
    <StyledLabel >{props.value}
    <br></br></StyledLabel>
);

export default Label;
