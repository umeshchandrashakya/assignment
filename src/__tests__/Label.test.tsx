import * as React from "react";
import { render, shallow, mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import Label from "../components/atoms/Label";

describe("<Label />", () => {
    it("should renders without crashing", () => {
        const component = render(<MemoryRouter><Label value=
            'Username' /></MemoryRouter>);
        expect(component).toMatchSnapshot();
    });

    it("should render with props provided", () => {
        const wrapper = mount(<Label value=
            'Username' />);
        wrapper.update();
        const text = wrapper.find('label').text();
        expect(text).toEqual('Username');
    });
});
