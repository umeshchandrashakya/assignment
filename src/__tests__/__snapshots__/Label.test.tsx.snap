// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> should renders without crashing 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "sc-bdVaJa lncEqa",
    },
    "children": Array [
      Object {
        "data": "Username",
        "next": Object {
          "attribs": Object {},
          "children": Array [],
          "name": "br",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
      Object {
        "attribs": Object {},
        "children": Array [],
        "name": "br",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "data": "Username",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "text",
        },
        "type": "tag",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
    ],
    "name": "label",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
